src/MDI/mdi_engine.cpp

```diff
@@ -18,322 +18,1662 @@
   MDI_Register_command("@FORCES", "EXIT");
 
-  // node at POST_INTEGRATE location in timestep
-
-  MDI_Register_node("@COORDS");
-  MDI_Register_command("@COORDS", "<@");
-  MDI_Register_command("@COORDS", "<CELL");
-  MDI_Register_command("@COORDS", "<CELL_DISPL");
-  MDI_Register_command("@COORDS", "<CHARGES");
-  MDI_Register_command("@COORDS", "<COORDS");
-  MDI_Register_command("@COORDS", "<ENERGY");
-  MDI_Register_command("@COORDS", "<FORCES");
-  MDI_Register_command("@COORDS", "<KE");
-  MDI_Register_command("@COORDS", "<LABELS");
-  MDI_Register_command("@COORDS", "<MASSES");
-  MDI_Register_command("@COORDS", "<NATOMS");
-  MDI_Register_command("@COORDS", "<PE");
-  MDI_Register_command("@COORDS", "<TYPES");
-  MDI_Register_command("@COORDS", ">CELL");
-  MDI_Register_command("@COORDS", ">CELL_DISPL");
-  MDI_Register_command("@COORDS", ">COORDS");
-  MDI_Register_command("@COORDS", ">FORCES");
-  MDI_Register_command("@COORDS", ">+FORCES");
-  MDI_Register_command("@COORDS", "@");
-  MDI_Register_command("@COORDS", "@COORDS");
-  MDI_Register_command("@COORDS", "@DEFAULT");
-  MDI_Register_command("@COORDS", "@FORCES");
-  MDI_Register_command("@COORDS", "EXIT");
+  // node at END_OF_STEP location in timestep
+  // only used if fix MDI/ENGINE is instantiated
+
+  MDI_Register_node("@ENDSTEP");
+  MDI_Register_command("@ENDSTEP", "<@");
+  MDI_Register_command("@ENDSTEP", "<ENERGY");
+  MDI_Register_command("@ENDSTEP", "<FORCES");
+  MDI_Register_command("@ENDSTEP", "<KE");
+  MDI_Register_command("@ENDSTEP", "<PE");
+  MDI_Register_command("@ENDSTEP", "<STRESS");
+  MDI_Register_command("@ENDSTEP", "@");
+  MDI_Register_command("@ENDSTEP", "@DEFAULT");
+  MDI_Register_command("@ENDSTEP", "@COORDS");
+  MDI_Register_command("@ENDSTEP", "@FORCES");
+  MDI_Register_command("@ENDSTEP", "@ENDSTEP");
+  MDI_Register_command("@ENDSTEP", "EXIT");
+}
 
-  // if the mdi/engine fix is not already present, add it now
+/* ----------------------------------------------------------------------
+   run MD simulation
+   either for NITERATE steps or one step at a time
+   latter is controlled by driver
+---------------------------------------------------------------------- */
 
-  bool added_mdi_engine_fix = false;
-  auto fixes = modify->get_fix_by_style("mdi/engine");
-  if (fixes.size() == 0) {
-    added_mdi_engine_fix = true;
-    mdi_fix = (FixMDIEngine *) modify->add_fix("MDI_ENGINE_INTERNAL all mdi/engine");
-  } else
-    mdi_fix = (FixMDIEngine *) fixes.front();
+void MDIEngine::mdi_md()
+{
+  // create or update system if requested prior to @INIT_MD
+
+  int flag_create = flag_natoms | flag_types;
+  if (flag_create) create_system();
+  else {
+    if (flag_cell || flag_cell_displ) adjust_box();
+    if (flag_charges) adjust_charges();
+    if (flag_coords) adjust_coords();
+    if (flag_velocities) adjust_velocities();
+  }
 
-  // check that LAMMPS is setup as a compatible MDI engine
+<<<<<<< HEAD
```

Comment:
@sjplimp you have committed here an unresolved merge conflict. This causes the code to fail compiling.